core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7384,3 +7384,48 @@ wasm_runtime_is_underlying_binary_freeable(WASMModuleCommon *const module)
 
     return true;
 }
+
+/*
+ * compare with a bigger type set in `wasm_value_type_size_internal()`,
+ * this function will only cover global value type, function's param
+ * value type and function's result value type.
+ *
+ * please feel free to add more if there are more requirements
+ */
+bool
+is_valid_value_type(uint8 type)
+{
+    if (/* I32/I64/F32/F64, 0x7C to 0x7F */
+        (type >= VALUE_TYPE_F64 && type <= VALUE_TYPE_I32)
+#if WASM_ENABLE_GC != 0
+        /* reference types, 0x65 to 0x70 */
+        || wasm_is_type_reftype(type)
+#elif WASM_ENABLE_REF_TYPES != 0
+        || (type == VALUE_TYPE_FUNCREF || type == VALUE_TYPE_EXTERNREF)
+#endif
+#if WASM_ENABLE_SIMD != 0
+#if (WASM_ENABLE_WAMR_COMPILER != 0) || (WASM_ENABLE_JIT != 0)
+        || type == VALUE_TYPE_V128 /* 0x7B */
+#endif
+#endif
+    )
+        return true;
+    return false;
+}
+
+bool
+is_valid_func_type(const WASMFuncType *func_type)
+{
+    unsigned i;
+    for (i = 0; i < func_type->param_count; i++) {
+        if (!is_valid_value_type(func_type->types[i]))
+            return false;
+    }
+
+    for (i = 0; i < func_type->result_count; i++) {
```

Comment:
Had better `for (i = 0; i < func_type->param_count + func_type->result_count; i++) { ... }`?

Reply:
Sure. no need to separate params and results.